examples/icmp_response.rs

```diff
@@ -0,0 +1,278 @@
+    }
+
+    /* initialize rx_batch_size and packet metadata */
+    let rx_batch_size: u32 = 64;
+    let mut packets: Vec<pv::Packet> = Vec::with_capacity(rx_batch_size as usize);
+
+    while !term.load(Ordering::Relaxed) {
+        let received: u32 = pv::pv_receive(&mut nic, &mut packets, rx_batch_size);
+
+        if received > 0 {
+            let processed: u32 =
+                process_packets(&mut nic, &mut packets, received, &src_mac_address);
+            let sent: u32 = pv::pv_send(&mut nic, &mut packets, processed);
+
+            if sent == 0 {
+                for i in (0..processed).rev() {
+                    pv::pv_free(&mut nic, &mut packets, i as usize);
+                }
+            }
+        }
+    }
+
+    pv::pv_close(nic);
+    println!("PV END");
+}
+
+// only packet to be replied is remaining in packets. other packets are freed.
+fn process_packets(
+    nic: &mut pv::Nic,
+    packets: &mut Vec<pv::Packet>,
+    batch_size: u32,
+    src_mac_address: &MacAddr,
+) -> u32 {
+    let mut processed = 0;
+    for i in (0..batch_size as usize).rev() {
+        match get_protocol(&packets[i]) {
+            Protocol::ARP => {
+                if make_arp_response_packet(src_mac_address, &mut packets[i]).is_ok() {
+                    processed += 1;
+                } else {
+                    pv::pv_free(nic, packets, i);
+                }
+            }
+            Protocol::ICMP => {
+                if make_icmp_response_packet(&mut packets[i]).is_ok() {
+                    processed += 1;
+                } else {
+                    pv::pv_free(nic, packets, i);
+                }
+            }
+            _ => {
+                pv::pv_free(nic, packets, i);
+            }
+        }
+    }
+    processed
+}
+
+// analyze what kind of given packet
+fn get_protocol(packet: &pv::Packet) -> Protocol {
+    if is_icmp(packet) {
+        Protocol::ICMP
+    } else if is_arp_req(packet) {
+        Protocol::ARP
+    } else {
+        Protocol::OTHER
+    }
+}
+
+fn set_eth(packet: &mut MutableEthernetPacket) {
+    let dest_mac_addr: MacAddr = packet.get_source();
+    let src_mac_addr: MacAddr = packet.get_destination();
+
+    packet.set_destination(dest_mac_addr);
+    packet.set_source(src_mac_addr);
+}
+
+fn set_ipv4(packet: &mut MutableIpv4Packet) {
+    let dest_ip_addr: Ipv4Addr = packet.get_source();
+    let src_ip_addr: Ipv4Addr = packet.get_destination();
+
+    packet.set_destination(dest_ip_addr);
+    packet.set_source(src_ip_addr);
+    packet.set_flags(0x00);
+    packet.set_checksum(ipv4::checksum(&packet.to_immutable()));
+}
+
+fn set_icmp(packet: &mut MutableIcmpPacket) {
+    packet.set_icmp_type(IcmpTypes::EchoReply);
+    packet.set_checksum(icmp::checksum(&packet.to_immutable()));
+}
+
+fn make_icmp_response_packet(packet: &mut pv::Packet) -> Result<(), String> {
+    let mut buffer = packet.get_buffer_mut();
+
+    let mut eth_pkt = MutableEthernetPacket::new(&mut buffer).unwrap();
```

Comment:
`make_icmp_response_packet()` returns `Result<(), String>`.
So, `unwrap()` should be returning `String` when the value type is `Err`

please, try to use `match` keyword and make error handling

Reply:
The team leader sugessted me to change this code by using `match` keyword in the last friday.
Maybe, the following code would be the code:

`let eth_pkt;
match MutableEthernetPacket::new(&mut buffer) {
    Some(a) => {
        eth_pkt = a;
    },
    None => {
        return Err(String::from(
            "buffer size is less than the minimum required packet size",
        ));
    }
}`

please try this code

Reply:
```rs
let eth_pkt = match MutableEthernetPacket::new(&mut buffer) {
  Some(a) => a,
  None => { return Err(String::from( "buffer size is less than the minimum required packet size", )); },
}
```